### PR TITLE
Create CheckTimer

### PR DESCRIPTION
**Pull request by beshur**
- [x] wrap timer into a class
- [x] bundle content scripts
- [x] bundle background scripts
- [x] add watch because of bundling

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -26,9 +26,31 @@ module.exports = function(grunt) {
             }
         },
 
+        rollup: {
+            options: {
+                format: "iife"
+            },
+            dist: {
+                files: {
+                    "./dist/index.js": "./src/content/main.js",
```
Comment (endway) [inline Gruntfile.js]:
```suggestion
                    "./dist/content.js": "./src/content/main.js",
```

File: manifest.json [hunk 2]
```diff
@@ -32,13 +32,15 @@
     "*://*/*"
    ],
    "js": [
-    "src/content/index.js"
+    "dist/index.js"
```
Comment (endway) [inline manifest.json]:
```suggestion
    "dist/content.js"
```

File: src/content/CheckTimer.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        }
+      }, _this.options.delay);
+    }
+    check();
+  }
+
+  stop() {
+    clearTimeout(this.timer);
+  }
+};
```
Comment (endway) [inline src/content/CheckTimer.js]: newline

File: src/content/CheckTimer.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    this.options = options;
+    this.timer = null;
+
+    this.start = this.start.bind(this);
+    this.stop = this.stop.bind(this);
+  }
+
+  start() {
+    let _this = this;
+    let check = function() {
```
Comment (endway) [inline src/content/CheckTimer.js]: Let's move it to a separate method:
```
check = () => { ... };
```
That way we no longer need *_this*

File: .gitignore [hunk 5]
```diff
@@ -1,2 +1,3 @@
 /node_modules
-/builds
\ No newline at end of file
+/builds
+/dist
```
Comment (endway) [inline .gitignore]: newline

File: src/content/CheckTimer.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    callback (function) - function to call on delay
+    recursive (boolean) - restart timer after callback is fired
+
+   */
+  constructor(options) {
+    this.options = options;
+    this.timer = null;
+
+    this.check = this.check.bind(this);
+    this.start = this.start.bind(this);
```
Comment (endway) [inline src/content/CheckTimer.js]: only check method needs bind

File: src/content/CheckTimer.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      }
+    }, this.options.delay);
+  }
+
+  start() {
+    this.check();
+  }
+
+  stop() {
+    clearTimeout(this.timer);
```
Comment (endway) [inline src/content/CheckTimer.js]:
```suggestion
    if (!this.timer) {
        return;
    }

    clearTimeout(this.timer);
```

File: src/content/CheckTimer.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    this.options = options;
+    this.timer = null;
+
+    this.check = this.check.bind(this);
+    this.start = this.start.bind(this);
+    this.stop = this.stop.bind(this);
+  }
+
+  check() {
+    clearTimeout(this.timer);
```
Comment (endway) [inline src/content/CheckTimer.js]:
```suggestion
    this.stop();
```

File: src/content/CheckTimer.js (hunk 6 above)
Comment (beshur) [reply src/content/CheckTimer.js]: No, `start` and `stop` are used by external classes

**Pull request merged**